src/hardware/pcspeaker.cpp

```diff
@@ -29,50 +29,60 @@
 #include <algorithm>
 #include <cassert>
 #include <cmath>
+#include <vector>
 
 #include "mixer.h"
 #include "timer.h"
 #include "setup.h"
 #include "support.h"
 #include "pic.h"
 
-#define SPKR_ENTRIES 1024
-#define SPKR_VOLUME  40000
+// constants
+constexpr auto SPKR_ENTRIES = 1024;
+constexpr auto SPKR_VOLUME = 40000;
 
-#define SPKR_FILTER_QUALITY 100
-#define SPKR_OVERSAMPLING   32
-#define SPKR_CUTOFF_MARGIN  0.2 // must be greater than 0.0
-#define SPKR_FILTER_WIDTH   (SPKR_FILTER_QUALITY * SPKR_OVERSAMPLING)
-#define SPKR_HIGHPASS       0.999 // FIXME: should be selected based on sampling rate
+constexpr auto SPKR_FILTER_QUALITY = 100;
+constexpr auto SPKR_OVERSAMPLING = 32;
+constexpr auto SPKR_CUTOFF_MARGIN = 0.2f; // must be greater than 0.0f
+constexpr auto SPKR_FILTER_WIDTH = (SPKR_FILTER_QUALITY * SPKR_OVERSAMPLING);
+constexpr auto SPKR_HIGHPASS = 0.999f; // FIXME: should be selected based on
+                                       // sampling rate
+constexpr auto ms_per_pit_tick = 1000.0f / PIT_TICK_RATE;
+constexpr auto pi_f = static_cast<float>(M_PI);
```

Comment:
Should we use UPPER_CASE for constants consistently?

Reply:
I'm not sure.. it's certainly a norm to all-caps `#define` values, but I've preferred switxhing them to lowercase when making them bonafide variables using constexpr's. 
So I think in this case, I think it's fair to lower-case them too, now that they're constexpr's. 

I personally don't like all caps just in general; they interrupt casual reading giving an impression of being off limits/untouchable/correct.

Reply:
Will switch to lower-case; thanks @johnnovak .

Also thinking I should take this refactoring the rest of the way (similar to Tandy, GUS, PS1), with a class, and cleanup init and close routines (without the control class baggage).